src/Microsoft.AspNet.Razor/Chunks/Generators/TagHelperDirectiveChunkGenerator.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Razor.Parser.SyntaxTree;
+
+namespace Microsoft.AspNet.Razor.Chunks.Generators
+{
+    public abstract class TagHelperDirectiveChunkGenerator : SpanChunkGenerator
```

Comment:
Why duplicate the logic in each of the 3 chunk generators? Then ontop of that have to do the following:

```C#
TagHelperDirectiveType directiveType;
string directiveText;
if (span.ChunkGenerator is AddTagHelperChunkGenerator)
{
    directiveType = TagHelperDirectiveType.AddTagHelper;
    directiveText = ((AddTagHelperChunkGenerator)span.ChunkGenerator).GetDirectiveText(span);
}
else if (span.ChunkGenerator is RemoveTagHelperChunkGenerator)
{
    directiveType = TagHelperDirectiveType.RemoveTagHelper;
    directiveText = ((RemoveTagHelperChunkGenerator)span.ChunkGenerator).GetDirectiveText(span);
}
else if (span.ChunkGenerator is TagHelperPrefixDirectiveChunkGenerator)
{
    directiveType = TagHelperDirectiveType.TagHelperPrefix;
    directiveText = ((TagHelperPrefixDirectiveChunkGenerator)span.ChunkGenerator).GetDirectiveText(span);
}
else
{
    // Not a chunk generator that we're interested in.
    return;
}
```